pkg/daemon/update.go

```diff
@@ -423,10 +427,32 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 	}
 
 	if dn.nodeWriter != nil {
-		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStaged", "Changes to OS staged")
+		// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+		// Remove this when we can ensure all events are sent before exiting.
+		ref, err := reference.GetReference(scheme.Scheme, dn.node)
+		if err == nil {
+			t := metav1.Time{Time: time.Now()}
+			event := &corev1.Event{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      fmt.Sprintf("%v.%x", ref.Name, t.UnixNano()),
+					Namespace: metav1.NamespaceDefault,
+				},
+				Reason:         "OSUpdateStaged",
+				Type:           corev1.EventTypeNormal,
+				Message:        "Changes to OS staged",
+				FirstTimestamp: t,
```

Comment:
Looking at our docs: https://docs.openshift.com/container-platform/4.10/rest_api/metadata_apis/event-events-k8s-io-v1.html

It seems we should probably be using eventTime instead? and first/last timestamps are deprecated

Reply:
This component is still using cores events API. The docs above is the new events API. Here is the [core](https://docs.openshift.com/container-platform/4.10/rest_api/metadata_apis/event-v1.html) events API docs.

Reply:
Eventually, this component must move away from cores events API before its removed from client go but for now, I think its good I remain consistent.